src/utils/app/zip-import-export.ts

```diff
@@ -0,0 +1,140 @@
+import { Conversation } from '@/src/types/chat';
+import { DialFile } from '@/src/types/files';
+import { FolderInterface } from '@/src/types/folder';
+
+import { AttachmentToUpload } from '@/src/store/import-export/importExport.reducers';
+
+import { constructPath, triggerDownload } from './file';
+import { prepareConversationsForExport } from './import-export';
+
+import JSZip from 'jszip';
+
+interface GetZippedFile {
+  files: DialFile[];
+  conversations: Conversation[];
+  folders: FolderInterface[];
+}
+export async function getZippedFile({
+  files,
+  conversations,
+  folders,
+}: GetZippedFile) {
+  const zip = new JSZip();
+  const filesPromises = files.map(async (file) => {
+    const fileResult = await fetch(
+      `api/files/file/${constructPath(file.absolutePath, file.name)}`,
+    );
+
+    zip.file(`res/${file.id}`, fileResult.blob());
```

Comment:
SHould it return zip.file result?

Reply:
No. It should not. It adds a file to the zip file.